tests/unit/Service/DefaultBoardServiceTest.php

```diff
@@ -0,0 +1,191 @@
+ *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+namespace OCA\Deck\Service;
+
+use OCA\Deck\Db\Card;
+use OCA\Deck\Db\Board;
+use OCA\Deck\Db\Stack;
+use OCA\Deck\Db\BoardMapper;
+use OCA\Deck\Service\DefaultBoardService;
+use OCA\Deck\Service\BoardService;
+use OCA\Deck\Service\StackService;
+use OCA\Deck\Service\CardService;
+use OCP\IConfig;
+use OCP\IL10N;
+use \Test\TestCase;
+
+class DefaultBoardServiceTest extends TestCase {
+	
+	/** @var DefaultBoardService */
+	private $service;	
+
+	/** @var BoardService */
+	private $boardService;
+
+	/** @var StackService */
+	private $stackService;
+
+	/** @var CardService */
+	private $cardService;
+
+	/** @var BoardMapper */
+	private $boardMapper;	
+
+	/** @var IConfig */
+	private $config;
+
+	private $l10n;
+	
+	private $userId = 'admin';	
+
+	public function setUp() {
+		parent::setUp();
+		$this->boardMapper = $this->createMock(BoardMapper::class);
+		$this->boardService = $this->createMock(BoardService::class);
+		$this->stackService = $this->createMock(StackService::class);
+		$this->cardService = $this->createMock(CardService::class);
+		$this->config = $this->createMock(IConfig::class);
+		$this->l10n = $this->createMock(IL10N::class);		
+
+		$this->service = new DefaultBoardService(
+			$this->l10n,
+			$this->boardMapper,
+			$this->boardService,
+			$this->stackService,
+			$this->cardService,
+			$this->config
+		);		
+	}
+
+	public function testCheckFirstRunCaseTrue() {
+		$appName = 'deck';
+		$userBoards = [];
+
+		$this->config->expects($this->once())
+			->method('getUserValue')
+			->willReturn('yes');
+		
+		$this->boardMapper->expects($this->once())
+			->method('findAllByUser')
+			->willReturn($userBoards);
+
+		$this->config->expects($this->once())
+			->method('setUserValue');			
+
+		$result = $this->service->checkFirstRun($this->userId, $appName);
+		$this->assertEquals($result, true);
+	}
+
+	public function testCheckFirstRunCaseFalse() {
+		$appName = 'deck';
+		$board = new Board();
+		$board->setTitle('Personal');
+		$board->setOwner($this->userId);
+		$board->setColor('000000');
+
+		$userBoards = [$board];
+
+		$this->config->expects($this->once())
+			->method('getUserValue')
+			->willReturn('no');
+
+		$this->boardMapper->expects($this->once())
+			->method('findAllByUser')
+			->willReturn($userBoards);
+
+		$result = $this->service->checkFirstRun($this->userId, $appName);
+		$this->assertEquals($result, false);
+	}
+
+	public function testCreateDefaultBoard() {	
+		$title = 'Personal';		
+		$color = '000000';
+		$boardId = 5;
+		
+		$board = new Board();
+		$board->setId($boardId);
+		$board->setTitle($title);
+		$board->setOwner($this->userId);
+		$board->setColor($color);
+		$this->boardService->expects($this->once())
+			->method('create')
+			->willReturn($board);
+
+		$stackToDoId = '123';		
+		$stackToDo = $this->assembleTestStack($this->l10n->t('To do'), $stackToDoId, $boardId);		
```

Comment:
If you mock the t call on l10n with something like:
```
$this->l10n->expects($this->any())
	->method('t')
	->willReturnCallback(function($text) { return $text } );
```

you can just call $this->assembleTestStack($this->l10n->t('

Reply:
Great, just submitted the revision, once your build server finishes it's thing you should be good to go :).